### PR TITLE
Add replication strategy support (#1)

### PR DESCRIPTION
**Pull request by markglh**
- Added support for configurable replication strategy when creating the keyspace. A valid config will look like:

Simple Strategy

```
pillar.faker {
    development {
        cassandra-seed-address: "127.0.0.1"
        cassandra-keyspace-name: "pillar_development"
        replicationStrategy: "SimpleStrategy"
        replicationFactor: 0
    }
}
```

Network Topology Strategy

```
pillar.faker {
    development {
        cassandra-seed-address: "127.0.0.1"
        cassandra-keyspace-name: "pillar_development"
        replicationStrategy: "NetworkTopologyStrategy"
        replicationFactor: [
            {dc1: 2},
            {dc2: 3}
        ]
    }
}
```
- Added CassandraUnit to enable testing & CI without a local Cassandra:
  https://github.com/jsevellec/cassandra-unit
- Tweaked README
- Set the default `consistencyLevel` to `QUORUM`. This ensures that the migrations are immediately consistent (both read and write will use `QUORUM`). High consistency here is essential to ensure that we always read the latest migrations and don't try to re-apply. Additionally, performance is not a concern as this are executed as a one-off. 

We deliberately didn't make the consistency level configurable as we don't believe it adds value, there isn't a valid reason to use a lower consistency level than `QUORUM` here.

### CODE REVIEW

**Comment by markglh** [comment]
Credit for the initial work goes to @j-potts

File: .travis.yml [hunk 1]
```diff
@@ -4,5 +4,3 @@ jdk:
    - oraclejdk8
 scala:
    - 2.11.8
-services:
```
Comment (markglh) [inline .travis.yml:7]: not required now we're using cassandra-unit

File: src/test/resources/cassandra_embedded.yaml [hunk 2]
```diff
@@ -0,0 +1,596 @@
+####################################################################
```
Comment (markglh) [inline src/test/resources/cassandra_embedded.yaml:1]: This is required for cassandra-unit so we can generate a random port

File: src/test/scala/de/kaufhof/pillar/PillarCommandLineAcceptanceSpec.scala [hunk 3]
```diff
@@ -1,64 +1,79 @@
...
 import com.datastax.driver.core.exceptions.InvalidQueryException
 import com.datastax.driver.core.querybuilder.QueryBuilder
+import com.typesafe.config.ConfigFactory
 import de.kaufhof.pillar.cli.App
-import org.scalatest.matchers.ShouldMatchers
-import org.scalatest.{BeforeAndAfter, FeatureSpec, GivenWhenThen}
+import org.scalatest.{BeforeAndAfter, FeatureSpec, GivenWhenThen, Matchers}
+
+class PillarCommandLineAcceptanceSpec extends FeatureSpec
+  with CassandraSpec
```
Comment (markglh) [inline src/test/scala/de/kaufhof/pillar/PillarCommandLineAcceptanceSpec.scala:10]: This gives us an embedded cassandra

File: src/test/scala/de/kaufhof/pillar/PillarCommandLineAcceptanceSpec.scala [hunk 4]
```diff
@@ -1,64 +1,79 @@
...
   val keyspaceName = "pillar_acceptance_test"
+  val environment = "acceptance_test"
+  val dataStoreName = "faker"
+
+  //create a configuration pointing to the embedded cassandra, falling back to the default configuration
+  lazy val config = ConfigFactory.parseString(
+    s"""
+       |pillar.$dataStoreName {
+       |    $environment {
+       |        cassandra-port: $port
```
Comment (markglh) [inline src/test/scala/de/kaufhof/pillar/PillarCommandLineAcceptanceSpec.scala:25]: we need this to configure the port as it's random

**Comment by markglh** [comment]
@lichtsprung we also have another pr almost ready to go which build on this one, adding quorum consistency for the migrations. This is essential when running against several nodes.

**Comment by lichtsprung** [comment]
@markglh Awesome! I will try to review both PRs this Friday :)

**Comment by markglh** [comment]
@lichtsprung that's great, thanks! We'll raise the second PR later today then so it's ready for you. It builds upon this one so probably best to review + merge this one before reviewing the second

**Comment by markglh** [comment]
I've merged the consistency level changes into this pr as they kind of go hand in hand and are fairly trivial.
See here: https://github.com/Galeria-Kaufhof/pillar/pull/9/commits/2a956146c6ed6d3137ba59ecb3752718c03882a9

**Comment by lichtsprung** [comment]
@markglh I agree with that you don't want to use a lower consistency level than QUORUM but what if you need LOCAL_QUORUM consistency?

**Comment by markglh** [comment]
If it's a single DC then we'd effectively have `LOCAL_QUORUM` anyway, however with multiple DC's my thinking was that you would always want these schema changes replicating across all of them as they're obviously pretty important. I was actually toying with `EACH_QUORUM` but I'm not sure if that's necessary.

Are you thinking we may want `LOCAL_QUORUM` under some circumstances? @lichtsprung

**Comment by markglh** [comment]
To elaborate on the multiple DC thing, my thinking was that having inconsistent keyspaces across DC's could potentially be a major issue if applications start running before they're consistent

**Comment by lichtsprung** [comment]
@markglh Not sure. Was just wondering if there might be a (good) reason for using some variant of `QUORUM`.

I think I'd prefer `EACH_QUORUM` in a multi DC setting though.

**Comment by lichtsprung** [comment]
...Not sure if this is really an issue or necessary for a multiple DCs.

@markglh  We can merge this version and see if anyone complains that they need some higher consistency level for whatever reason.

**Comment by markglh** [comment]
@lichtsprung works for me. If feedback is that we want this configurable we can make the change. Would you like it changing to EACH_QUORUM before the merge?

**Comment by lichtsprung** [comment]
@markglh Let me ask our ops team. If they prefer something other than quorum, we need to make it configurable.

**Comment by lichtsprung** [comment]
@markglh I'll publish a new version on Monday.

**Comment by markglh** [comment]
@lichtsprung that's great, thanks!!

**Pull request merged**